flink-table/flink-table-runtime/src/test/java/org/apache/flink/table/runtime/operators/join/RandomSortMergeInnerJoinTest.java

```diff
@@ -126,15 +126,11 @@ public void test() throws Exception {
         match(expectedMatchesMap, transformToBinary(join(operator, input1, input2)));
 
         // assert that each expected match was seen
-        for (Map.Entry<Integer, Collection<Match>> entry : expectedMatchesMap.entrySet()) {
-            assertThat(entry.getValue().isEmpty())
-                    .as("Collection for key " + entry.getKey() + " is not empty")
-                    .isTrue();
-        }
+        assertThat(expectedMatchesMap).allSatisfy((i, e) -> assertThat(e).isNotEmpty());
```

Comment:
should assert isEmpty, or use noneSatisfy

Reply:
now you inverted both and its again wrong :D

We were previously checking that all contained collections are empty. So either allSatisfy(isEmpty) or nonSatisfy(isNotEmpty)